lib/validator/common.go

```diff
@@ -0,0 +1,88 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validator
+
+import (
+	"fmt"
+
+	"github.com/projectcalico/libcalico-go/lib/api"
+	"github.com/projectcalico/libcalico-go/lib/api/unversioned"
+	"github.com/projectcalico/libcalico-go/lib/errors"
+	"github.com/projectcalico/libcalico-go/lib/scope"
+)
+
+// ValidateKeyFieldsExist is used to validate the Key fields in a resource
+// to ensure that all necessary fields are present.
+// This is split from the validator so it can be used conditionally
+// depending on the command.
+func ValidateKeyFieldsExist(resource unversioned.Resource) error {
+	return ValidateMetadataFieldsExist(resource.GetResourceMetadata())
+}
+
+// ValidateMetadataFieldsExist is used to validate the Resource Metadata to ensure
+// that all necessary fields are present.
+// This is split from the validator so it can be used conditionally
+// depending on the command.
+func ValidateMetadataFieldsExist(rm unversioned.ResourceMetadata) error {
+	switch metadata := rm.(type) {
+	case api.BGPPeerMetadata:
+		if metadata.PeerIP.IP == nil {
+			return errors.ErrorInsufficientIdentifiers{Name: "peerIP"}
+		}
+		if metadata.Scope != scope.Global && metadata.Node == "" {
+			return errors.ErrorInsufficientIdentifiers{Name: "node"}
+		}
+	case api.HostEndpointMetadata:
+		if metadata.Node == "" {
+			return errors.ErrorInsufficientIdentifiers{Name: "node"}
+		}
+		if metadata.Name == "" {
+			return errors.ErrorInsufficientIdentifiers{Name: "name"}
+		}
+	case api.IPPoolMetadata:
+		if metadata.CIDR.IP == nil {
+			return errors.ErrorInsufficientIdentifiers{Name: "cidr"}
+		}
+	case api.NodeMetadata:
+		if metadata.Name == "" {
+			return errors.ErrorInsufficientIdentifiers{Name: "name"}
+		}
+	case api.PolicyMetadata:
+		if metadata.Name == "" {
+			return errors.ErrorInsufficientIdentifiers{Name: "name"}
+		}
+	case api.ProfileMetadata:
+		if metadata.Name == "" {
+			return errors.ErrorInsufficientIdentifiers{Name: "name"}
+		}
+	case api.WorkloadEndpointMetadata:
+		if metadata.Node == "" {
+			return errors.ErrorInsufficientIdentifiers{Name: "node"}
+		}
+		if metadata.Orchestrator == "" {
+			return errors.ErrorInsufficientIdentifiers{Name: "orchestrator"}
+		}
+		if metadata.Workload == "" {
+			return errors.ErrorInsufficientIdentifiers{Name: "workload"}
+		}
+		if metadata.Name == "" {
+			return errors.ErrorInsufficientIdentifiers{Name: "name"}
+		}
+	default:
+		return fmt.Errorf("Unexpected resource metadata: %s", metadata)
```

Comment:
I think this should be Fatal - it indicates some code that hasn't but should have been written.